jeison.el

```diff
@@ -198,12 +198,12 @@ PATH is a `list' of keys we should consequently find in JSON and
 proceed with a nested JSON further on."
   ;; read JSON from string into an alist and proceed
   (jeison--read-internal type
-                           (funcall (or (and (bufferp inp-source)
-                                             #'jeison--read-json-buffer)
-                                        (and (stringp inp-source)
-                                             #'jeison--read-json-string)
-                                        #'identity)
-                                    inp-source)
+                         (funcall (cond
```

Comment:
I also meant to use cond, to get rid of `funcall`, it's really unnecessary here

Reply:
No, `funcall` is needed. The first argument of an eval-ed list must be a function. `cond` is not a function, unless you meant rewriting as:

```emacs-lisp
(cond ((bufferp inp-source) (jeison--read-buffer inp-source))
       ((stringp inp-source) (jeison--read-string inp-source))
       (t inp-source))
```

Whiich is a bit repetitive. `funcall` makes it more compact and more of the Lisp style of code (FWIW, I've been coding in CL since ~1985.)

Reply:
Yeah, I meant rewriting it that way simply because it's one less level of indirection. I like simple dumb code. Because of this I don't like using old code or other code as a benchmark of what is considered normal in terms of style. In many cases, old code tends to be too smart and too convoluted for its own good.

Reply:
You're taking my Lisp-foo magic away!